### PR TITLE
Issue 49987: Add labkey.experiment.lineage to Rlabkey

### PR DESCRIPTION
**Pull request by cnathe**
#### Rationale
Add labkey.experiment.lineage API to support getting parent/child relationships for exp objects by LSID
https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=49987

#### Related PRs
- https://github.com/LabKey/testAutomation/pull/2012

#### Changes
- Add labkey.experiment.lineage function and doc info with example

### CODE REVIEW

File: Rlabkey/R/labkey.experiment.R
Comment (labkey-klum) [inline Rlabkey/R/labkey.experiment.R:224]: interesting, I didn't know what an `atomic` vector was.

File: Rlabkey/man/labkey.experiment.lineage.Rd
Comment (labkey-klum) [inline Rlabkey/man/labkey.experiment.lineage.Rd]: Did you intend to comment this out?

**Review by labkey-klum — APPROVED** [review]

File: Rlabkey/R/labkey.experiment.R
Comment (cnathe) [reply Rlabkey/R/labkey.experiment.R:224]: this was the suggestion when I googled how to check for a param as a vector. Apparently an R list will return is.vector() true for some reason

File: Rlabkey/man/labkey.experiment.lineage.Rd
Comment (cnathe) [reply Rlabkey/man/labkey.experiment.lineage.Rd]: I was leaving it in as an example of another option, but I suppose it doesn't hurt to uncomment it. Done

**Comment by cnathe** [comment]
The CRAN package submission page is unavailable until Aug 16, so I guess I'll wait until then to finish this merge.

**Pull request merged**
